### PR TITLE
Fixup typo in SQA documents checklist

### PR DESCRIPTION
**Pull request by cticenhour**
This stood out in a recent SQA check I performed, so I thought I should just fix it. 

This PR also adds a simple deprecation capability for SQA documents, to ease this sort of renaming transition for applications. The extension will now process the right document, if the old/new name pairs are provided, and issue a warning that the new name needs to be transitioned to in the `sqa_reports.yml` file. This warning will display on the SQA check status page for the application as well as on the command line. 

Refs #17191

### CODE REVIEW

**Review by GiudGiud — DISMISSED** [review]
superior spelling

**Comment by GiudGiud** [comment]
looks like you'll need to patch the apps. 
Ugh this will conflict with my PR and my app patches. I ll live

**Comment by cticenhour** [comment]
Yeah....considering this is in all the SQA docs, I will likely need to break next to get this done, so this might not go in this week.

File: python/moosesqa/check_documents.py [hunk 1]
```diff
@@ -33,6 +33,16 @@ def check_documents(documents, file_list=None, **kwargs):
         root = mooseutils.git_root_dir()
         file_list = mooseutils.git_ls_files(root, recurse_submodules=False)
 
+    # If deprecation adjustments occurred, report them as warnings
+    if len(deprecations) != 0:
+        for old_name, new_name in deprecations.items():
+            log_key = "log_" + new_name
+            logger.setLevel(log_key, logging.WARNING)
+            msg = "The document name '{}' has been deprecated, please use '{}' instead.".format(old_name, new_name)
```
Comment (cticenhour) [inline python/moosesqa/check_documents.py:41]:
```suggestion
            msg = "The document name '{}' has been deprecated, please use '{}' instead in sqa_reports.yml.".format(old_name, new_name)
```

File: python/moosesqa/get_documents.py [hunk 2]
```diff
@@ -36,6 +36,9 @@ def __init__(self, **kwargs):
                  'user_manual',
                  'theory_manual']
 
+# List of deprecated document names to ease application patching (provided in (deprecated_name, new_name) pairs)
```
Comment (cticenhour) [inline python/moosesqa/get_documents.py:39]:
```suggestion
# List of deprecated document names to ease application patching (provided in 'deprecated_name' : 'new_name' pairs)
```

**Review by cticenhour — COMMENTED** [review]
For later

**Comment by cticenhour** [comment]
OpenMPI test failure unrelated (not initializing correctly for one module test). Pinging @GiudGiud for another review with the new deprecation feature.

**Review by GiudGiud — APPROVED** [review]
so is this new deprecation API going to allow to deprecate the heat_conduction ones?

or is there still an issue because the apps still refer to a 'heat_conduction' section in their sqa docs

I guess we ll see once this is merged

**Comment by cticenhour** [comment]
This would only affect the SQA document report, (so the stuff configured using `sqa_reports.yml`). So there would still be an issue for MooseDocs more broadly.

**Pull request merged**